docs/modules/ROOT/pages/Java_Case_Study.adoc

```diff
@@ -0,0 +1,430 @@
+= Java Case Study: Netty
+
+++++
+<script>
+gtag('config', 'AW-16649289906');
+</script>
+++++
+
+This page compares using Mill to Maven, using the [Netty Network Server](https://github.com/netty/netty)
+codebase as the example. Netty is a large, old codebase. 500,000 lines of Java, written by
+over 100 contributors across 15 years, split over 47 subprojects, with over 10,000 lines of
+Maven `pom.xml` configuration alone. By porting it to Mill, this case study should give you
+an idea of how Mill compares to Maven in larger, real-world projects.
+
+To do this, we have written a Mill `build.sc` file for the Netty project. This can be used
+with Mill to build and test the various submodules of the Netty project without needing to
+change any other files in the repository:
+
+- ???[Netty `build.sc` file]
+
+== Completeness
+
+The Mill build for Netty is not 100% complete, but it covers most of the major parts of Netty:
+compiling Java, compiling and linking C code via JNI, running JUnit tests and some integration
+tests using H2Spec. All 47 Maven subprojects are modelled using Mill.
+
+The goal of this exercise is not to be 100% feature complete enough to replace the Maven build
+today. It is instead meant to provide a realistic comparison of how using Mill in a large,
+complex project compares to using Maven.
+
+== Performance
+
+The Mill build for Netty is much more performant than the default Maven build. This applies to
+most workflows:
+
+=== Sequential Clean Compile
+
+```bash
+$ time ./mvnw -DskipTests  -Dcheckstyle.skip -Denforcer.skip=true clean install
+
+$ time ./mill clean __.compile
```

Comment:
I think these are not fair comparisions. We either also run `jar` in Mill or we don't `install` in Maven. 

```suggestion
$ time ./mvnw -DskipTests  -Dcheckstyle.skip -Denforcer.skip=true clean compile

$ time ./mill clean __.compile
```

Running just compile sometimes work in a Maven reactor build, sometimes not. Need to check.

Alternatively, we need to run `jar` in Mill.

```suggestion
$ time ./mvnw -DskipTests  -Dcheckstyle.skip -Denforcer.skip=true clean install

$ time ./mill clean __.jar
```

Reply:
The thing is, in Mill the common workflow is running `compile`, while in Maven the equivalent workflow is running `install`. AFAICT, this is because Maven uses `~/.m2/local` for passing dependencies between subprojects, while Mill doesn't.

Part of the reason Mill is faster is because it does less work, but I don't think that makes the comparison unfair. In the end, nobody runs `compile` in Maven or `jar` in Mill. `mvn install` and `mill compile` are equivalent user workflows